.github/workflows/_linux-test.yml

```diff
@@ -124,6 +124,7 @@ jobs:
           SHARD_NUMBER: ${{ matrix.shard }}
           NUM_TEST_SHARDS: ${{ matrix.num_shards }}
           PR_BODY: ${{ github.event.pull_request.body }}
+          PR_LABELS: ${{ toJson(github.event.pull_request.labels.*.name) }}
```

Comment:
With this approach, note that the labels from `github.event.pull_request.labels` is static extracted from the PR when the request is sent to the runner.  So if you attach the label to the PR after the jobs have started, it would make no change until you update the PR.  Re-running won't help.

Reply:
adding on to this, nikita is correct in that this wont work for trunk or inductor since github.event.pull_request is only available to pull_request triggered events

Reply:
As an alternative, we could tweak https://github.com/pytorch/pytorch/blob/master/.github/scripts/filter_test_configs.py#L57 script that runs before all  tests jobs.  It dynamically queries all the labels from the PR, and set the test matrix.  May be it could be extended to influence how the test is performed, i.e. have a `keep_going` tag added

Reply:
I think I'm going to go the filter_test_configs.py approach